app_dart/lib/src/model/ci_yaml/ci_yaml.dart

```diff
@@ -18,14 +19,35 @@ class CiYaml {
   });
 
   /// The underlying protobuf that contains the raw data from .ci.yaml.
-  final pb.SchedulerConfig config;
+  pb.SchedulerConfig config;
 
   /// The [RepositorySlug] that [config] is from.
   final RepositorySlug slug;
 
   /// The git branch currently being scheduled against.
   final String branch;
 
+  /// Load [currentConfigYaml] (and optionally [totConfigYaml] ) to [pb.SchedulerConfig] and validate the dependency graph.
+  /// where totYaml means tip of tree Yaml.
+  CiYaml.schedulerConfigFromYaml(YamlMap? currentConfigYaml,
+      {YamlMap? totConfigYaml, bool ensureBringupTargets = false})
```

Comment:
If we have to pass in `totConfigYaml`, I would just remove `ensureBringupTarget` as run the check whenever the tot config is passed